_posts/2023-08-23-even-friendlier-sql.md

```diff
@@ -0,0 +1,596 @@
+
+## Function chaining
+
+Many SQL blogs advise the use of CTE’s instead of subqueries. Among other benefits, they are much more readable. Operations are compartmentalized into discrete chunks and they can be read in order top to bottom instead of forcing the reader to work their way inside out. 
+
+DuckDB enables the same interpretability improvement for every scalar function! Use the dot operator to chain functions together, just like in Python. The prior expression in the chain is used as the first argument to the subsequent function. 
+
+```sql
+SELECT 
+     ('Make it so')
+          .UPPER()
+          .string_split(' ')
+          .list_aggr('string_agg','.')
+          .concat('.') AS im_not_messing_around_number_one;
+```
+
+| im_not_messing_around_number_one |
+|:---|
+| MAKE.IT.SO.                      |
+
+Now compare that with the old way...
+
+```sql
+SELECT 
+     concat(
+          list_aggr(
+               string_split(
+                    UPPER('Make it stop'),
+               ' '),
+          'string_agg','.'),
+     '.') AS oof;
+```
+
+|      oof      |
+|:---|
+| MAKE.IT.STOP. |
+
+## Union by name
+
+DuckDB aims to blend the best of databases and dataframes. This new syntax is inspired by the [concat function in Pandas](https://pandas.pydata.org/docs/reference/api/pandas.concat.html). Rather than vertically stacking tables based on column position, columns are matched by name and stacked accordingly. Simply replace `UNION` with `UNION BY NAME` or `UNION ALL` with `UNION ALL BY NAME`. 
+
+For example, we had to add some new alien species proverbs in The Next Generation:
+```sql
+SELECT 
+     'Revenge is a dish best served cold' AS klingon_proverb 
+UNION ALL BY NAME 
+SELECT 
+     'You will be assimilated' AS borg_proverb,
+     'If winning is not important, why keep score?' AS klingon_proverb ;
+```
+
+|               klingon_proverb                |      borg_proverb       |
+|:---|:---|
+| Revenge is a dish best served cold           | NULL                    |
+| If winning is not important, why keep score? | You will be assimilated |
+
+This approach has additional benefits. As seen above, not only can tables with different column orders be combined, but so can tables with different numbers of columns entirely. This is helpful as schemas migrate, and is particularly useful for DuckDB’s [multi-file reading capabilities](https://duckdb.org/docs/data/multiple_files/combining_schemas.html#union-by-name). 
+
+## Insert by name
+
+Another common situation where column order is strict in SQL is when inserting data into a table. Either the columns must match the order exactly, or all of the column names must be repeated in two locations within the query. 
+
+Instead, add the keywords `BY NAME` after the table name when inserting. Any subset of the columns in the table in any order can be inserted.
+
+```sql
+INSERT INTO proverbs BY NAME 
```

Comment:
the proverbs table does not exist at this point, it should be created with

```sql
CREATE TABLE proverbs AS
    SELECT 
         'Revenge is a dish best served cold' AS klingon_proverb 
    UNION ALL BY NAME 
    SELECT 
         'You will be assimilated' AS borg_proverb,
         'If winning is not important, why keep score?' AS klingon_proverb;
```

Reply:
Another great catch!